src/Authenticator/PasetoAuthenticator.php

```diff
@@ -0,0 +1,196 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authentication\Authenticator;
+
+use ArrayObject;
+use Authentication\Identifier\IdentifierInterface;
+use Exception;
+use ParagonIE\Paseto\JsonToken;
+use ParagonIE\Paseto\Keys\AsymmetricSecretKey;
+use ParagonIE\Paseto\Keys\SymmetricKey;
+use ParagonIE\Paseto\Parser;
+use ParagonIE\Paseto\ProtocolCollection;
+use ParagonIE\Paseto\ProtocolInterface;
+use ParagonIE\Paseto\Purpose;
+use Psr\Http\Message\ServerRequestInterface;
+use RuntimeException;
+
+/**
+ * PASETO Authenticator
+ *
+ * Authenticates an identity based on a PASETO token.
+ *
+ * @link https://github.com/paragonie/paseto
+ * @link https://github.com/paseto-standard/paseto-spec
+ */
+class PasetoAuthenticator extends TokenAuthenticator
+{
+    public const LOCAL = 'local';
+    public const PUBLIC = 'public';
+
+    /**
+     * @inheritDoc
+     */
+    protected $_defaultConfig = [
+        'header' => 'Authorization',
+        'queryParam' => 'token',
+        'tokenPrefix' => 'bearer',
+        'returnPayload' => true,
+        'version' => null,
+        'purpose' => null,
+        'secretKey' => null,
+    ];
+
+    /**
+     * Payload data.
+     *
+     * @var object|null
+     */
+    protected $payload;
+
+    /**
+     * @inheritDoc
+     */
+    public function __construct(IdentifierInterface $identifier, array $config = [])
+    {
+        parent::__construct($identifier, $config);
+
+        if ($this->getConfig('version') !== null && !$this->getConfig('version') instanceof ProtocolInterface) {
+            throw new RuntimeException(
+                'PASETO `version` config must be an instance of ' . ProtocolInterface::class
+            );
+        }
+
+        if (!in_array($this->getConfig('purpose'), [self::PUBLIC, self::LOCAL])) {
+            throw new RuntimeException('PASETO `purpose` config must one of: local or public');
+        }
+
+        if (empty($this->getConfig('secretKey'))) {
+            if (!class_exists(\Cake\Utility\Security::class)) {
```

Comment:
Don't know how to test this, thinking this should be marked as code coverage ignore.